### ISSUE TITLE
Android Compatibility / Testing Thoughts

### ISSUE THREAD

**Issue by bleege** [mapbox/mapbox-gl-native#604]
**tl;dr - Bulk Of Android Devices Will Be On Older Versions Of Android API And Likely On Samsung Hardware**

It's super tempting to focus in on latest / greatest version of Android that Google releases, I just want to make sure that we're careful here to remember that Android version marketshare moves at a glacier pace compared to iOS.  In other words while iOS 8 is already on over 50% iPhones / iPads / iPods according to [Mixpanel](https://mixpanel.com/trends/#report/ios_8/from_date:-29,report_unit:day,to_date:0) since it was released in September, Android 5 isn't even registering on Google's own marketshare dashboard yet.  The *Any versions with less than 0.1% distribution are not shown.* is the key.  For comparison [here's how](https://github.com/mapbox/mapbox-android-sdk/issues/53) the Raster SDK dealt with it back in March.

![20141115-android-marketshare](https://cloud.githubusercontent.com/assets/989928/5058680/0c2d589c-6cc0-11e4-9a0c-8b0e63e20389.png)

Android Marketshare
https://developer.android.com/about/dashboards/index.html

iOS Official Marketshare (Behind Great Wall of Apple)
https://developer.apple.com/support/appstore/

**Test Device Strategy Thoughts**

Nexus devices are fun.  They also never amount to any serious marketshare as they tend to marketed at / bought be developers like us.  The vast majority of Android users (aka potential Mapbox Android GL end users) end up buying their phones through the carriers, which means they end up buying Samsung, LG, Motorola, or HTC.  Check that... [they end up buying Samsung](http://www.comscore.com/Insights/Market-Rankings/comScore-Reports-September-2014-US-Smartphone-Subscriber-Market-Share).

Samsung sells like a [zillion versions of their devices](http://www.samsung.com/us/mobile/cell-phones/all-products).  So what to do?  The good news is that they tend to focus around core product ranges like phone, phablet, and tablet and release "flagship versions" every year like Apple does.  Right now that means:

* phone - [Samsung Galaxy S5](http://www.samsung.com/gr/microsite/galaxys5/)
* phablet - [Samsung Galaxy Note 4](http://www.samsung.com/us/mobile/galaxy-note/)
* tablet - [Samsung Galaxy Tab](http://www.samsung.com/us/mobile/galaxy-tab/)

So if Mapbox is thinking about building out a small cache of Android devices to develop and test with these Samsung devices would be a good place to start IMHO.  Then rely on people's personal devices which will naturally vary across hardware vendor and Android API version to do more breadth testing.

/cc @incanus @ljbade @springmeyer

**Comment by bleege** [comment]
Revisiting this topic as the @mapbox/mobile team is much bigger today than when I originally wrote this and we should all be thinking about this as we build and more importantly test Mapbox Android tech products for the general public.  While Samsung may not have as dominant of marketshare as it did in the Fall 2014 it still has more of it's phones in the hands of the general public than Google does with Nexus devices.  In other words Nexus devices continue to be reference devices that Google intended them to be for developers and not indicative of the general public device marketshare.  It also means that as we acquire devices for our daily development and testing we should keep the need for diversity in mind and NOT all have Nexus devices as our primary development device.

Looking at a recent [comScore analysis of the markeshare from April 2016](https://www.comscore.com/Insights/Rankings/comScore-Reports-February-2016-US-Smartphone-Subscriber-Market-Share), the usual OEM suspects (Samsung, LG, Motorola, and HTC) continue to ship the most Android devices in the USA (and likely Europe as well).

![screen shot 2016-07-22 at 9 17 27 am](https://cloud.githubusercontent.com/assets/989928/17060013/9bdc19e4-4fed-11e6-8118-a50699284670.png)

**Comment by jfirebaugh** [comment]
Is this actionable or can it be closed?

**Comment by bleege** [comment]
@jfirebaugh We're now working on this on implementing this internally to update and diversify our daily development devices.
